sphinxcontrib/katex.py

```diff
@@ -290,3 +310,251 @@ def get_node_equation_number(writer, node):
+        # select() on stdout to wait for the port/socket path without resorting
+        # to polling. However, select() is not supported for pipes on Windows.
+        startup_start = time.monotonic()
+        while True:
+            try:
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                remaining = startup_start + timeout - time.monotonic()
+                if remaining <= 0.0:
+                    raise cls.timeout_error(timeout)
+
+                with socket_timeout(sock, remaining):
+                    sock.connect((host, port))
+
+                break
+            except ConnectionRefusedError:
+                # The server is not up yet. Try again.
+                time.sleep(ONE_MILLISECOND)
+            except socket.timeout:
+                raise cls.timeout_error(timeout)
+
+        return process, sock
+
+    @classmethod
+    def start(cls):
+        rundir = Path(tempfile.mkdtemp(prefix="pelican_katex"))
+
+        if os.name == "posix":
+            process, sock = cls.start_server_process(rundir, STARTUP_TIMEOUT)
+        else:
+            # Non-unix systems (i.e. Windows) do not support unix
+            # domain sockets for IPC, so we use network sockets.
+            process, sock = cls.start_network_socket(rundir, STARTUP_TIMEOUT)
+
+        server = KaTeXServer(rundir, process, sock)
+
+        # Clean up after ourselves when skphinx is done. I don't want to register
+        # signal handlers here.
+        atexit.register(KaTeXServer.terminate, server)
+
+        return server
+
+    @classmethod
+    def get(cls):
+        """Get the current render server or start one"""
+        if cls.KATEX_SERVER is None:
+            cls.KATEX_SERVER = KaTeXServer.start()
+
+        return cls.KATEX_SERVER
+
+    def __init__(self, rundir, process, sock):
+        self.rundir = rundir
+        self.process = process
+        self.sock = sock
+
+        # 100KB should be large enough even for big equations
+        self.buffer = bytearray(100 * 1024)
+
+    def terminate(self):
+        """Terminate the render server and clean up"""
+        self.sock.close()
+        try:
+            self.process.terminate()
+            self.process.wait(timeout=self.STOP_TIMEOUT)
+        except TimeoutExpired:
+            self.process.kill()
+        shutil.rmtree(self.rundir)
+
+    def render(self, request, timeout=None):
+        # Configure timeouts
+        if timeout is not None:
+            start_time = time.monotonic()
+            self.sock.settimeout(timeout)
+        else:
+            self.sock.settimeout(None)
+
+        # Send the request
+        request_bytes = json.dumps(request).encode("utf-8")
+        length = len(request_bytes)
+        self.sock.sendall(self.LENGTH_STRUCT.pack(length))
+        self.sock.sendall(request_bytes)
+
+        # Read the amount of bytes we are about to receive
+        length = self.LENGTH_STRUCT.unpack(self.sock.recv(self.LENGTH_STRUCT.size))[0]
```

Comment:
This line is slightly too long and it would be nice to introduce a helping variable explaining the output of some of the steps, e.g.
```suggestion
        size = self.sock.recv(self.LENGTH_STRUCT.size)
        length = self.LENGTH_STRUCT.unpack(size)[0]
```
But maybe replace `size` with something more meaningful, as I don't know what it should be ;)